third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -2652,8 +2675,11 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
         maccess: &EA::ModuleAccess,
         generics: &Option<Vec<EA::Type>>,
         fields: &EA::Fields<T>,
-        mut field_translator: impl FnMut(&mut Self, &Type, &T) -> S,
-    ) -> Option<(QualifiedInstId<StructId>, Vec<S>)> {
+        init: S,
+        // used to fold results when iterating over the fields
+        // the args are: translator, acc, field_name, exp_idx, def_idx, field_ty, value
+        mut combine: impl FnMut(&mut Self, &mut S, &Symbol, usize, usize, &Type, &T),
```

Comment:
This is too complex. A lambda with 6 arguments is not readable. Besides it is not documented what all those things mean. Perhaps put this into a struct (or part of it, like `FieldInfo{field_name,, exp_idx, def_idx, field_ty}`?